FreeRTOS-Plus/Source/FreeRTOS-IoT-Libraries-LTS-Beta2/c_sdk/platform/freertos/transport/src/wolfSSL_freertos.c

```diff
@@ -0,0 +1,435 @@
+                    pHostName,
+                    pNetworkCredentials ) );
+        returnStatus = TLS_TRANSPORT_INVALID_PARAMETER;
+    }
+    else if( ( pNetworkCredentials->pRootCa == NULL ) )
+    {
+        LogError( ( "pRootCa cannot be NULL." ) );
+        returnStatus = TLS_TRANSPORT_INVALID_PARAMETER;
+    }
+
+    /* Establish a TCP connection with the server. */
+    if( returnStatus == TLS_TRANSPORT_SUCCESS )
+    {
+        socketStatus = Sockets_Connect( &( pNetworkContext->tcpSocket ),
+                                        pHostName,
+                                        port,
+                                        receiveTimeoutMs,
+                                        sendTimeoutMs );
+
+        if( socketStatus != 0 )
+        {
+            LogError( ( "Failed to connect to %s with error %d.",
+                        pHostName,
+                        socketStatus ) );
+            returnStatus = TLS_TRANSPORT_CONNECT_FAILURE;
+        }
+    }
+
+    /* Initialize tls. */
+    if( returnStatus == TLS_TRANSPORT_SUCCESS )
+    { 
+        returnStatus = initTLS();
+    }
+
+    /* Perform TLS handshake. */
+    if( returnStatus == TLS_TRANSPORT_SUCCESS )
+    {
+        returnStatus = tlsSetup( pNetworkContext, pHostName, pNetworkCredentials );
+    }
+
+    /* Clean up on failure. */
+    if( returnStatus != TLS_TRANSPORT_SUCCESS )
+    {
+        if( pNetworkContext->tcpSocket != FREERTOS_INVALID_SOCKET )
+        {
+            FreeRTOS_closesocket( pNetworkContext->tcpSocket );
+        }
+    }
+    else
+    {
+        LogInfo( ( "(Network connection %p) Connection to %s established.",
+                   pNetworkContext,
+                   pHostName ) );
+    }
+
+    return returnStatus;
+}
+
+/*-----------------------------------------------------------*/
+
+void TLS_FreeRTOS_Disconnect( NetworkContext_t * pNetworkContext )
+{
+    WOLFSSL* pSsl = pNetworkContext->sslContext.ssl;
+
+    /* shutdown an active TLS connection */
+    wolfSSL_shutdown(pSsl);
+
+    /* cleanup WOLFSSL object */
+    wolfSSL_free(pSsl);
+    pNetworkContext->sslContext.ssl = NULL;
+
+    /* Call socket shutdown function to close connection. */
+    Sockets_Disconnect(pNetworkContext->tcpSocket);
+
+    /* free WOLFSSL_CTX object*/
+    WOLFSSL_CTX* pCtx = pNetworkContext->sslContext.ctx;
+    wolfSSL_CTX_free(pCtx);
+    pNetworkContext->sslContext.ctx = NULL;
+
+    wolfSSL_Cleanup();
+}
+
+/*-----------------------------------------------------------*/
+
+int32_t TLS_FreeRTOS_recv( NetworkContext_t * pNetworkContext,
+                           void * pBuffer,
+                           size_t bytesToRecv )
+{
+    int32_t tlsStatus = 0;
+    int  iResult = 0;
+    WOLFSSL* pSsl = pNetworkContext->sslContext.ssl;
+    
+    iResult = wolfSSL_read( pSsl, pBuffer, bytesToRecv );
+
+    /* return: >0 the number of bytes to read, 0 means connection closed, <0 error */
+    if (iResult <= 0)
```

Comment:
I think it would be easier to read, if the switch statement was a part of the if statements